common/go-utils/src/inv.rs

```diff
@@ -103,20 +87,18 @@ pub fn list_upstream_artifacts() -> Result<Vec<Artifact>, String> {
         .map_err(|e| e.to_string())?
         .iter()
         .flat_map(|release| &release.files)
-        .filter(|file| !file.sha256.is_empty())
-        .filter_map(|file| {
-            file.target_arch()
-                .filter(|target| target == ARCH)
-                .map(|target| {
-                    Version::parse_go(&file.version)
-                        .map(|version| Artifact {
-                            go_version: file.version.clone(),
-                            semantic_version: version,
-                            architecture: target,
-                            sha_checksum: file.sha256.clone(),
-                        })
-                        .map_err(|e| e.to_string())
+        .filter(|file| !file.sha256.is_empty() && file.os == "linux" && file.arch == "amd64")
```

Comment:
i think it would be nicer if the association between the `GoFile` variants of `os` and `arch` and how we represent our supported target `os` and `arch` was more explicit but maybe that's overkill right now.

Reply:
I've added explicit mapping and error handling to ensure that only supported variants of upstream `os` and `arch` values are added to the inventory: https://github.com/heroku/buildpacks-go/pull/229/files#diff-093f94a1f107af2f526a22b23862bc48ded092574cb775569a20214d221f9b01R80-R108